### ISSUE TITLE
RabbitMQ: tracing headers not propagated to reply of @RabbitListener

### ISSUE THREAD

**Issue by pzapasni** [spring-cloud/spring-cloud-sleuth#1752]
In my app I'm using ``spring-cloud-starter-sleuth:2.2.5`` with ``spring-boot-starter-amqp:2.2.7``. I have a ``@RabbitListener`` annotated method which accepts a message, processes it, and returns a response:

```java
@Service
@RequiredArgsConstructor
@Slf4j
@Transactional
public class MyListener {

    private final MyService service;

    @RabbitListener(queues = "${amqp.queue.name}")
    public MyEvent handleCommand(@Payload @Valid MyCommand command) {
        log.info("Received command {}", command);
        return service.handleCommand(command);
    }
}
```

If the incoming message has the ``b3`` header, the tracing information is present throughout the processing of the message. However, when the response message from that processing is being returned to RabbitMQ, it does not contain the ``b3`` header, which breaks the tracing.

I've investigated this issue a bit, and it seems to me that the only instrumentation for the listener (or rather ``SimpleRabbitListenerContainerFactory``) is being done through the ``SleuthRabbitBeanPostProcessor`` and ``TracingRabbitListenerAdvice``, however I couldn't find any code there that would take care of injecting the header into the response message. Is my assumption correct and this is a missing feature, or did I miss something?

**Comment by marcingrzejszczak** [comment]
I wonder if that's not a problem in Brave. Can you file an issue there please?

**Comment by pzapasni** [comment]
Issue created

**Comment by OlgaMaciaszek** [comment]
Closing the issue. If it turns out the problem is in Sleuth, we'll reopen it.

**Comment by pzapasni** [comment]
Issue fixed in Brave v5.12.7
